src/mg/PAMI/Client/Impl/ClientImpl.php

```diff
@@ -443,11 +439,14 @@ public function close()
 	 */
 	public function __construct(array $options)
 	{
-        if (isset($options['log4php.properties'])) {
-            \Logger::configure($options['log4php.properties']);
-        }
-        $this->_logger = \Logger::getLogger('Pami.ClientImpl');
-	    $this->_host = $options['host'];
+		$this->setLogger(
+			new Log4phpAdapter(
```

Comment:
I added this to keep full backwards compatibility. If we require people to inject the logger, we are breaking that, which makes for unhappy users.

Personally, I usually try to adhere quite strictly to [semver](http://semver.org/), because it guarantees BC for minor releases. In that case, another options is to *do* require this in the constructor, and then to bump PAMI to version 2.0.0. But that seems a bit overkill to me.

As an aside: I don't see much difference between hardcoding `\Logger` and hardcoding `\Log4phpAdapter` :smile: